### PR TITLE
Add config to only lobotomize villagers that have been traded with

### PR DESCRIPTION
**Pull request by 2stinkysocks**
Added a config (defaults to old behavior) that makes villagers only get lobotomized if they've been traded with at least once (their trades are locked).

### CODE REVIEW

File: patches/server/0307-Add-config-to-only-lobotomize-if-villagers-have-been.patch [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+             interval *= 2;
+         }
+         if (this.level().getGameTime() % interval == 0) {
+             // offset Y for short blocks like dirt_path/farmland
+-            this.isLobotomized = !canTravelFrom(BlockPos.containing(this.position().x, this.getBoundingBox().minY + 0.0625D, this.position().z));
++            boolean canLobotomize = true;
++            if(shouldCheckForTradeLocked) {
++                canLobotomize = this.getVillagerXp() != 0;
++            }
++            this.isLobotomized = canLobotomize && !canTravelFrom(BlockPos.containing(this.position().x, this.getBoundingBox().minY + 0.0625D, this.position().z));
```
Comment (granny) [inline patches/server/0307-Add-config-to-only-lobotomize-if-villagers-have-been.patch]: inline the canLobotomize check into the `this.isLobotomized = ...` line.

```
this.isLobotomized = (shouldCheckForTradeLocked && this.getVillagerXp() != 0) && ...
```

**Review by granny — CHANGES_REQUESTED** [review]
Merge this patch into the lobotomize patch, feel free to add yourself as a co-author.

File: patches/server/0307-Add-config-to-only-lobotomize-if-villagers-have-been.patch (hunk 1 above)
Comment (2stinkysocks) [reply patches/server/0307-Add-config-to-only-lobotomize-if-villagers-have-been.patch]: condensed the logic, yours was a bit off when I double checked it. here's a truth table I used to check my work:

```
C = only lobotomize traded with
X = has no xp
E = existing lobotomize condition

F = is lobotomized


(C -> !X) ^ E

(!C v !X) ^ E

!(C ^ X) ^ E


C  X  E    F
0  0  0    0    !(0 ^ 0) ^ 0  0
0  0  1    1    !(0 ^ 0) ^ 1  1
0  1  0    0    !(0 ^ 1) ^ 0  0
0  1  1    1    !(0 ^ 1) ^ 1  1
1  0  0    0    !(1 ^ 0) ^ 0  0
1  0  1    1    !(1 ^ 0) ^ 1  1
1  1  0    0    !(1 ^ 1) ^ 0  0
1  1  1    0    !(1 ^ 1) ^ 1  0
```

**Review by granny — APPROVED** [review]
LGTM! All that's left is creating a PR to PurpurDocs documenting the new option :) Once that's done both PRs will be merged later today (PST)

**Comment by 2stinkysocks** [comment]
https://github.com/PurpurMC/PurpurDocs/pull/88

**Pull request merged**